examples/derive-patterns/join-inner.flow.yaml

```diff
@@ -9,26 +9,50 @@ collections:
       required: [Key]
     key: [/Key]
 
-    derivation:
-      register:
-        schema: schema.yaml#Join
-        initial: {}
-
-      transform:
-        fromInts:
-          source: { name: patterns/ints }
+    derive:
+      using:
+        sqlite:
+          migrations:
+            - |
+              create table join_state (
+                key text not null primary key,
+                -- Stores the left hand side of the join
+                lhs integer,
+                -- Stores the right hand side of the join, using a JSON array
+                -- since this is a one-to-many join
+                rhs json
+              );
+      transforms:
+        - name: fromInts
+          source: patterns/ints
           shuffle: { key: [/Key] }
-          update: { lambda: typescript }
-          publish: { lambda: typescript }
+          lambda: |
+            insert into join_state (key, lhs) values ($Key, $Int)
+            on conflict (key) do update set lhs = lhs + $Int;
+            -- now emit the joined result
+            SELECT JSON_OBJECT(
```

Comment:
nit / not required: a pattern I've found quite nice when there are multiple lambdas (and would recommend to users), is to define a view in terms of join state which is the desired output of the derivation.

Then, across the multiple lambdas, do an upsert and then a `select * from output_view where key = $Key;`